lib/SILGen/SILGenFunction.cpp

```diff
@@ -457,12 +457,36 @@ void SILGenFunction::emitArtificialTopLevel(ClassDecl *mainClass) {
     CanType IUOptNSStringTy
       = ImplicitlyUnwrappedOptionalType::get(NSStringTy)->getCanonicalType();
 
+    // Look up UIApplicationMain.
+    // FIXME: Doing an AST lookup here is gross and not entirely sound;
+    // we're getting away with it because the types are guaranteed to already
+    // be imported.
+    ASTContext &ctx = getASTContext();
+    Module *UIKit = ctx.getLoadedModule(ctx.getIdentifier("UIKit"));
+    SmallVector<ValueDecl *, 1> results;
+    UIKit->lookupQualified(UIKit->getDeclaredType(),
+                           ctx.getIdentifier("UIApplicationMain"),
+                           NL_QualifiedDefault,
+                           /*resolver*/nullptr,
+                           results);
+    assert(!results.empty() && "couldn't find UIApplicationMain in UIKit");
+    assert(results.size() == 1 && "more than one UIApplicationMain?");
+
+    SILDeclRef mainRef{results.front(), ResilienceExpansion::Minimal,
+                       SILDeclRef::ConstructAtNaturalUncurryLevel,
+                       /*isForeign*/true};
+    auto UIApplicationMainFn = SGM.M.getOrCreateFunction(mainClass, mainRef,
+                                                         NotForDefinition);
+    assert(UIApplicationMainFn && "failed to lower UIApplicationMain to SIL");
```

Comment:
I don't think getOrCreateFunction() can return nullptr, and if it did you would crash on the next line anyway, so I suggest removing this assert.